clang/runtime/dpct-rt/include/math.hpp

```diff
@@ -242,6 +242,27 @@ inline unsigned unordered_compare_mask(const sycl::marray<T, 2> a,
       .as<sycl::vec<unsigned, 1>>();
 }
 
+template <typename T>
+inline T bfe(T source, uint32_t bit_start, uint32_t num_bits) {
+  if (!num_bits)
+    return 0;
```

Comment:
Is this special casing worth the branch?  
If num_bits is 0, the mask will be `(1 << 0) - 1 = 0`, and will return 0.  
I would expect the branch compared to the regular function path is not much faster, and this will slow down all the `num_bits != 0` cases.